test/spec/params.spec.ts

```diff
@@ -154,21 +154,125 @@ describe('params reached to function', () => {
     const fclient = new FunctionsClient(`http://localhost:${relay.container.getMappedPort(8081)}`)
 
     log('invoke mirror')
-    const customHeader = nanoid();
-    const validRegion = FunctionRegion.ApNortheast1;
+    const customHeader = nanoid()
+    const validRegion = FunctionRegion.ApNortheast1
 
     const { data, error } = await fclient.invoke<MirrorResponse>('mirror', {
       headers: {
         'custom-header': customHeader,
         Authorization: `Bearer ${apiKey}`,
       },
-      region: validRegion
+      region: validRegion,
     })
 
     log('assert no error')
-    expect(data).not.toBeNull()
+    const expected = {
+      url: 'http://localhost:8000/mirror',
+      method: 'POST',
+      headers: data?.headers ?? [],
+      body: '',
+    }
+    expect(data).toEqual(expected)
+    attach(
+      'check headers from function',
+      `expected to include: ${['custom-header', customHeader]}\n actual: ${JSON.stringify(
+        data?.headers
+      )}`,
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke with region overrides region in the client', async () => {
+    /**
+     * @feature headers
+     */
+    log('create FunctionsClient')
+    const fclient = new FunctionsClient(`http://localhost:${relay.container.getMappedPort(8081)}`, {
+      region: FunctionRegion.ApNortheast1,
+    })
+
+    log('invoke mirror')
+    const customHeader = nanoid()
+    const validRegion = FunctionRegion.ApSoutheast1
+
+    const { data, error } = await fclient.invoke<MirrorResponse>('mirror', {
+      headers: {
+        'custom-header': customHeader,
+        Authorization: `Bearer ${apiKey}`,
+      },
+      region: validRegion,
+    })
+
+    log('assert no error')
+    const expected = {
+      url: 'http://localhost:8000/mirror',
+      method: 'POST',
+      headers: data?.headers ?? [],
+      body: '',
+    }
+    expect(data).toEqual(expected)
+    attach(
+      'check headers from function',
+      `expected to include: ${['custom-header', customHeader]}\n actual: ${JSON.stringify(
+        data?.headers
+      )}`,
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke with region overrides region in the client', async () => {
```

Comment:
@mansueli can you please rename test to reflect the idea?

i just drafted these new ones, so would be great if all new tests will have meaningful names

Reply:
I've updated the names of the tests